src/common/utils/window.h

```diff
@@ -6,9 +6,10 @@
 
 #include <array>
 #include <optional>
+#include <functional>
 
 // Initializes and runs windows message loop
-inline int run_message_loop(const bool until_idle = false, const std::optional<uint32_t> timeout_seconds = {})
+inline int run_message_loop(const bool until_idle = false, const std::optional<uint32_t> timeout_seconds = {}, std::function<void(WORD, WORD)> onHotkey = ([](WORD modifiersMask, WORD vkCode) {}))
```

Comment:
Since the `run_message_loop` implementation is now very specific for the runner, let's move it to the runner project.

Reply:
Since the previous version was used by other code, let's leave the exiting common version untouched and create a new `RunMessageLoop` in the runner project.